pkg/dataobj/planner/physical/visitor.go

```diff
@@ -0,0 +1,13 @@
+package physical
+
+type Traversable interface {
+	Accept(Visitor) error
+}
+
+type Visitor interface {
+	VisitDataObjScan(*DataObjScan) error
+	VisitSortMerge(*SortMerge) error
+	VisitProjection(*Projection) error
+	VisitFilter(*Filter) error
+	VisitLimit(*Limit) error
+}
```

Comment:
One downside to this approach is that visitors have to implement each of these methods, even if they're not interested in every single node. As the number of nodes increases, implementing a Visitor requires more boilerplate. Also, each time we add a Node, we must update all Visitor implementations to compile again, even if they don't need to handle the new nodes. 

For that reason I usually prefer how Go does its [AST traversal](https://pkg.go.dev/go/ast#Walk), where one function gets recursively called and the Visitor takes the high-level `Node` interfae that the Visitor can do a type switch on. 

The choice is yours though; this is totally fine; it trades off ease of use for guarantees that every Visitor implementation is aware of all Node types at compile time.

Reply:
Let's keep it for now. If we see it is impractical, we can change it. It's not much of a change then.